sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -133,8 +143,15 @@ def read(
         """
         request_options = build_options(kwargs)
         response_hook = kwargs.pop('response_hook', None)
-        if populate_query_metrics is not None:
-            request_options["populateQueryMetrics"] = populate_query_metrics
+        populate_query_metrics = args[0] if args else kwargs.pop('populate_query_metrics', None)
+        if populate_query_metrics:
+            warnings.warn(
+                "the populate_query_metrics flag does not apply to this method and will be removed in the future",
+                UserWarning,
+            )
+        populate_partition_key_range_statistics = args[1] if args else kwargs.pop(
+            "populate_partition_key_range_statistics", None)
+        populate_quota_info = args[2] if args else kwargs.pop("populate_quota_info", None)
```

Comment:
This could potentially raise an IndexError - to be safe I think you'd need:
`... = args[2] if len(args) > 2 else ...`
(and similar for line 152)